mautic/mautic
# don't count hidden fields in progressive profiling fields limit

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | no
| New feature?                           | yes (more an enhancement)
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/278
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
This PR just extend the list for do not progressive profiling. At the moment it uses viewOnlyFields list. We need also use hidden field, then we create doNotProgressiveFields list, probably in future we can dispatch it and add support for plugins extend it.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create form with email, firstname, hidden field type (with default value)
3. Set progressive profiling to 1
![image](https://user-images.githubusercontent.com/462477/127889947-87f0dfda-b7f3-427c-b868-47edf4e4c9f8.png)
4. Submit form and see hidden field is not save/submitted
5. After PR hidden field is save, even If not pass progressive profiling limit.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10312"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
